tol_lab_share/processors/bioscan_pool_xp_to_traction_processor.py

```diff
@@ -16,41 +17,41 @@
 class BioscanPoolXpToTractionProcessor(BaseProcessor):
     """Processor for consumed messages exporting Pool XP tubes to Traction."""
 
-    def __init__(self, schema_registry: SchemaRegistry, config: Any):
+    def __init__(self, config: Any):
         """Prepare the initial internal state of the processor.
 
         Args:
-            schema_registry (SchemaRegistry): a schema registry for message parsing.
             config (Any): main configuration for the app.
         """
         LOGGER.debug("BioscanPoolXpToTractionProcessor::__init__")
 
-        self._schema_registry = schema_registry
         self._config = config
 
     @staticmethod
     def instantiate(
```

Comment:
Pretty sure I'm missing something here but, why are we creating a new function `instantiate` rather than just using the `__init__` function?

Reply:
Got it. It's part of the contract for the subclass. Didn't see `lab-share-lib` having the abstract method `instantiate`. 

All good.

Reply:
The initialiser is called by the lab-share-lib library so I defined the `BaseProcessor` there with a static method to initialise an instance.  You can't impose expectations of the arguments on the `__init__` method so when you create instances in `lab-share-lib` by calling the constructor with those arguments, there's no guarantee the classes you're instantiating will accept the arguments you want to pass.